### PR TITLE
Full rework: Type safe API with Mir

### PR DESCRIPTION
**Pull request by 9il**

### CODE REVIEW

File: integration/source/integration/api.d [hunk 1]
```diff
@@ -24,14 +24,14 @@ unittest {
...
-    database.insert(Measurement("cpu", ["tag1": "foo"], ["temperature": "68"]));
+    database.insert(Measurement("cpu", ["tag1": "foo"], ["temperature": 42]));
+    database.insert(Measurement("cpu", ["tag1": "foo"], ["temperature": 68]));
 
     {
         const response = database.query("SELECT * from cpu");
         const result = response.results[0];
         const series = result.series[0];
-        series.rows.length.shouldEqual(2);
+        series.rows.length.shouldEqual(1);
```
Comment (9il) [inline integration/source/integration/api.d:34]: The old implementation was wrong. 1 is the correct value because tags are the same and no timestamps.

**Comment by 9il** [comment]
@atilaneves did you make some changes? I have force pushed in miss them

**Comment by atilaneves** [comment]
> @atilaneves did you make some changes? I have force pushed in miss them

No.

**Comment by atilaneves** [comment]
1. There is no explanation of what this PR is trying to accomplish other than its title, which isn't nearly enough
2. The PR is too big even if it were multiple commits
3. The PR is one giant commit
4. It introduces new dependencies and doesn't justify them

**Pull request closed without merging**